services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -0,0 +1,164 @@
+func NewProducer(destinationConfig interface{}) (pubsub.Client, error) {
+	var config Config
+	ctx := context.Background()
+	jsonConfig, err := json.Marshal(destinationConfig)
+	if err != nil {
+		return pubsub.Client{}, fmt.Errorf("[GooglePubSub] Error while marshalling destination config :: %w", err)
+	}
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		return pubsub.Client{}, fmt.Errorf("[GooglePubSub] error  :: error in GooglePubSub while unmarshelling destination config:: %w", err)
+	}
+	var client *pubsub.Client
+	if config.Credentials != "" && config.ProjectId != "" {
+		client, err = pubsub.NewClient(ctx, config.ProjectId, option.WithCredentialsJSON([]byte(config.Credentials)))
+	}
+	if err != nil {
+		return pubsub.Client{}, err
+	}
+	return *client, nil
+}
+func Produce(jsonData json.RawMessage, producer interface{}, destConfig interface{}) (statusCode int, respStatus string, responseMessage string) {
+	parsedJSON := gjson.ParseBytes(jsonData)
+	pbs, ok := producer.(pubsub.Client)
+	ctx := context.Background()
+	if !ok {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: Could not create producer"
+		return 400, respStatus, responseMessage
+	}
+	var config Config
+	jsonConfig, err := json.Marshal(destConfig)
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error  :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	var data interface{}
+	if parsedJSON.Get("message").Value() != nil {
+		data = parsedJSON.Get("message").Value()
+	} else {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: message from payload not found"
+		return 400, respStatus, responseMessage
+	}
+	value, err := json.Marshal(data)
+
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error  :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	if parsedJSON.Get("topicId").Value() != nil {
+		topicIdString, ok := parsedJSON.Get("topicId").Value().(string)
+		if !ok {
+			respStatus = "Failure"
+			responseMessage = "[GooglePubSub] error :: Could not parse topic id to string"
+			logger.Error(responseMessage)
+			statusCode := 400
+			return statusCode, respStatus, responseMessage
+		}
+		if topicIdString == "" {
+			respStatus = "Failure"
+			responseMessage = "[GooglePubSub] error :: empty topic id string"
+			return 400, respStatus, responseMessage
+		}
+		topic := pbs.Topic(topicIdString)
+		result := topic.Publish(ctx, &pubsub.Message{Data: []byte(value)})
```

Comment:
// Publish publishes msg to the topic asynchronously. Messages are batched and
// sent according to the topic's PublishSettings. Publish never blocks.
//
// Publish returns a non-nil PublishResult which will be ready when the
// message has been sent (or has failed to be sent) to the server.
//
// Publish creates goroutines for batching and sending messages. These goroutines
// need to be stopped by calling t.Stop(). Once stopped, future calls to Publish
// will immediately return a PublishResult with an error.
func (t *Topic) Publish(ctx context.Context, msg *Message) *PublishResult {

Referring to the method's documentation, `Messages are batched and sent according to the topic's PublishSettings.` message will be sent to server once 10ms delayThreshold is reached according to the DefaultPublishSettings. This will limit our throughput. I think we should get rid of the delay threshold to immediately send our event, as we are not batching. 

Right after creating topic, we should set `topic.PublishSettings.DelayThreshold = 0`

Reply:
added this.